packages/core/data-transfer/src/strapi/queries/link.ts

```diff
@@ -180,10 +188,16 @@ export const createLinkQuery = (strapi: Strapi.Strapi) => {
 
       if (attribute.joinColumn) {
         const joinColumnName = attribute.joinColumn.name;
-
-        await connection(metadata.tableName)
-          .where('id', left.ref)
-          .update({ [joinColumnName]: right.ref });
+        if (trx) {
+          await connection(metadata.tableName)
+            .where('id', left.ref)
+            .transacting(trx)
+            .update({ [joinColumnName]: right.ref });
+        } else {
+          await connection(metadata.tableName)
+            .where('id', left.ref)
+            .update({ [joinColumnName]: right.ref });
+        }
```

Comment:
You could simplify with smth like that 

```suggestion
	const qb = connection(metadata.tableName)
          .where('id', left.ref)
          .update({ [joinColumnName]: right.ref });
        
        if (trx) {
          qb.transacting(trx);
        }
        
        await qb;
```